stdlib/nats/src/main/java/org/ballerinalang/nats/nativeimpl/Utils.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.ballerinalang.nats.nativeimpl;
+
+import org.ballerinalang.bre.Context;
+import org.ballerinalang.bre.bvm.BLangVMErrors;
+import org.ballerinalang.connector.api.BLangConnectorSPIUtil;
+import org.ballerinalang.connector.api.Resource;
+import org.ballerinalang.connector.api.Service;
+import org.ballerinalang.connector.api.Struct;
+import org.ballerinalang.model.types.BTypes;
+import org.ballerinalang.model.values.BError;
+import org.ballerinalang.model.values.BMap;
+import org.ballerinalang.model.values.BString;
+import org.ballerinalang.model.values.BValue;
+import org.ballerinalang.util.exceptions.BallerinaException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Utilities for producing and consuming via NATS sever.
+ */
+public class Utils {
+    private static final Logger LOGGER = LoggerFactory.getLogger(Utils.class);
+
+    /**
+     * Message which will be propagated.
+     */
+    private static final String MESSAGE = "message";
+
+    /**
+     * Constructs a BallerinaException and throw it.
+     *
+     * @param message error message which should be included.
+     * @param context context of the ballerina call.
+     * @param throwable the exception stacktrace.
+     */
+    public static void throwBallerinaException(String message, Context context, Throwable throwable) {
+        LOGGER.error(message, throwable);
```

Comment:
Shall we pass in the logger from the respective class in the method? Otherwise all logs would be from the Utils class?

Reply:
Same as described in https://github.com/ballerina-platform/ballerina-lang/pull/14669#discussion_r268972377

Reply:
Why log this at all? This is getting logged to the `ballerina.internal` log file. That's only meant for logging unexpected exceptions/abnormal conditions in the runtime which caused it to crash and there's nothing the user can do about it (i.e., a bug on our part). Throwing a BallerinaException means it's intended for the user. It will end up as a panic. These should not be logged in the `ballerina.internal` log file